pkg/api/mlflow/service/artifact/service.go

```diff
@@ -85,10 +87,20 @@ func (s Service) GetArtifact(
 		run.ArtifactURI, req.Path,
 	)
 	if err != nil {
-		return nil, api.NewInternalError(
+		msg := fmt.Sprintf(
 			"error getting artifact object for URI: %s",
 			filepath.Join(run.ArtifactURI, req.Path),
 		)
+		statError := regexp.MustCompile(storage.PathError)
+		isDirError := regexp.MustCompile(storage.IsDirError)
+		switch {
+		case statError.MatchString(err.Error()):
+			return nil, api.NewResourceDoesNotExistError(msg)
+		case isDirError.MatchString(err.Error()):
+			return nil, api.NewInvalidParameterValueError(msg)
```

Comment:
trying to get an existing directory should result in a `ResourceDoesNotExistError` too, like it already does in the S3 storage provider.

Reply:
i was thinking it does "exist" but we're requesting something we can't return in this case -- bad request. with S3, on the other hand, it does not actually exist.

Reply:
no artifact exists on that path — so 404
the reason I asked you to handle it is that if you do `os.Open` on a directory, it will actually work and give you a io.ReadCloser back, but will then of course fail with `EISDIR` later